share/spack/gitlab/cloud_pipelines/stacks/ml-linux-x86_64-cpu/spack.yaml

```diff
@@ -52,29 +75,71 @@ spack:
     - py-torchtext
     - py-torchvision
     - py-vector-quantize-pytorch
+    # GPU Specs
+    # PyTorch packages do not support Spack built ROCm
+    # so only configure CUDA and CPU versions
+    - matrix:
+      - - py-torch
+        - py-torch-nvidia-apex
+      - - $cuda_variants
```

Comment:
Will this propagate down through the DAG? For example, we need to build torchvision with torch+cuda~rocm, torch+rocm~cuda, and torch~cuda~rocm.

Reply:
Ah, I see. I didn't configure it that way, but I can. I missed that tensorflow/torch were backends to all of these packages, which makes sense.

I think I will write this so that we have something like this which specs all of the pacakges with the matrix of backends, xgboost would be another one I think.

Are there any packages with both Tensorflow and Torch dependencies? I did a quick search and didn't see any, but it is hard to read the graph with so many things in it.

```yaml
definitions:
- torch_packages:
  - ...
- torch_specs
  - matrix...
- tensorflow_packages:
  - ...
- tensorflow_specs
  - matrix...
- tensorflow_stack:
  - ...list of non-tensorflow dependent packages
  - matrix:
    - [$tensorflow_packages]
    - [^$tensorflow_specs]

- torch_stack:
  -   - ...list of non-torch dependent packages
  - matrix:
    - [$torch_packages]
    - [^$torch_specs]

specs:
- $torch_stack
- $tensorflow_stack
- $xgboost_stack
```

Reply:
It's not just torch/TF, it's possibly many packages in the DAG. I think you're better off tackling #34302 which would let us use `++cuda cuda_arch==80` for all packages regardless of whether or not they have these variants. Otherwise this is going to get very ugly and custom and difficult to maintain. I'm still dreaming of #32893 and #32894 but I think these are all viable alternatives.